### PR TITLE
refine pruner docs

### PR DESCRIPTION
**Pull request by WeiweiZhang1**
## Type of Change

documentation

### CODE REVIEW

**Review by YIYANGCAI — APPROVED** [review]

File: docs/source/pruning.md [hunk 1]
```diff
@@ -243,6 +243,24 @@ Regularization is a technique that discourages learning a more complex model and
 </a>
 </div>
 
+### Large Language Model Pruning
+
+To efficiently achieve pruning for Large Language Models (LLMs), we have implemented two post-training pruning methods that utilize different pruning patterns: **Retrain-free** (channel-wise) and **SparseGPT** (1x1 / 2:4 / 4:8).
```
Comment (wenhuach21) [inline docs/source/pruning.md:248]: add reference

Comment (wenhuach21) [inline docs/source/pruning.md:248]: 1x1/N:M

File: docs/source/pruning.md [hunk 2]
```diff
@@ -345,7 +362,99 @@ The following section exemplifies how to use hooks in user pass-in training func
...
+  pruning_configs = [
+      {  # config of a single pruner
+          "pruning_type": "retrain_free",
+          "pruning_scope": "global",
+          "op_names": [".fc", ".mlp"],  # MLP layer_names
+          "start_step": 1,
+          "end_step": 300,  # set end_step for Few shot pruning.
+          "excluded_op_names": ["lm_head"],  # A list of modules that would not be pruned.
+          "target_sparsity": 0.2,  # Target sparsity ratio of modules.
+          "pruning_frequency": 50,  # Frequency of applying pruning, The recommended setting is one fortieth of the pruning steps.
```
Comment (wenhuach21) [inline docs/source/pruning.md:383]: 1/40 may be not fit for this scenario

File: examples/pytorch/nlp/huggingface_models/language-modeling/pruning/eager/README.md [hunk 3]
```diff
@@ -23,11 +23,14 @@ The dataset will be downloaded automatically from the datasets Hub.
...
-There are pruning scripts for LLM sparse models (GPT-j, BLOOM, OPT, LLaMA etc). The sparse model can be obtained by modifying pruning parameters. [Pruning Scripts](https://github.com/intel/neural-compressor/tree/master/examples/pytorch/nlp/huggingface_models/language-modeling/pruning/eager/scripts/).
+Intel® Neural Compressor provides pruning and slimming capabilities for Large Language Models(LLMs) without the need for retraining. 
+Through experimental verification, we have found that pruning MLP layers using channel-wise patterns can achieve a sparsity of 10%-20% without sacrificing accuracy, thereby significantly speeding up inference by <1%. [Retrain-free Example](https://github.com/intel/neural-compressor/tree/master/examples/pytorch/nlp/huggingface_models/language-modeling/pruning/eager/run_clm_no_trainer.py).
```
Comment (wenhuach21) [inline examples/pytorch/nlp/huggingface_models/language-modeling/pruning/eager/README.md:30]: thereby significantly speeding up inference by <1%?    what's the meaning

File: neural_compressor/compression/pruner/README.md [hunk 4]
```diff
@@ -345,7 +363,100 @@ The following section exemplifies how to use hooks in user pass-in training func
...
+          "pruning_type": "retrain_free",
+          "pruning_scope": "global",
+          "op_names": [".fc", ".mlp"],  # MLP layer_names
+          "start_step": 1,
+          "end_step": 300,  # set end_step for Few shot pruning.
+          "excluded_op_names": ["lm_head"],  # A list of modules that would not be pruned.
+          "target_sparsity": 0.2,  # Target sparsity ratio of modules.
+          "pruning_frequency": 50,  # Frequency of applying pruning, The recommended setting is one fortieth of the pruning steps.
+          "pattern": "channelx1",  # Default pruning pattern.
+          "pruning_op_types": ["Linear"],
```
Comment (wenhuach21) [inline neural_compressor/compression/pruner/README.md:386]: currently we only support linear layer, I suggest to not expose purning_op_types  to users

**Review by wenhuach21 — APPROVED** [review]

**Pull request merged**